### PR TITLE
Fixes chowning of docker.sock

### PR DESCRIPTION
**Pull request by jnbnyc**
test, and see

### CODE REVIEW

**Comment by hypergig** [comment]
Cool, thanks @jnbnyc for the pr. I will take a look and test it out.

What does your docker environment look like? Are you using docker-machine, docker for mac, etc?

File: robo-jenkins/Dockerfile [hunk 1]
```diff
@@ -10,7 +10,10 @@ ENV JAVA_OPTS="-Djenkins.install.runSetupWizard=false" \
 
 # packages and sudoers
 USER root
-RUN apk add --no-cache docker sudo && echo "jenkins ALL=NOPASSWD: ALL" >> /etc/sudoers
+RUN apk add --no-cache docker sudo &&\
+    adduser jenkins users &&\
+    adduser jenkins docker &&\
```
Comment (hypergig) [inline robo-jenkins/Dockerfile]: does the docker apk add the docker group?

Comment (jnbnyc) [inline robo-jenkins/Dockerfile]: Yes. Typically the owner group is docker, I am using docker-machine for local development and I found that the mounted docker.sock volume was owned by users.

Comment (hypergig) [inline robo-jenkins/Dockerfile]: Not sure this works in a docker native environment...

**Comment by hypergig** [comment]
@ztidwell would you mind testing this on your rig to see if it works?  I feel like the docker group within the robo-jenkins container will not line up (gid) with your local one, if that is in fact how the user space is shared.

**Comment by hypergig** [comment]
ehh... not feeling hopeful

```
bash-4.3$ sudo groups
root bin daemon sys adm disk wheel floppy dialout tape video
bash-4.3$ 
```

**Comment by ztidwell** [comment]
@hypergig, no dice.  It looks like the gid is is creating for the docker daemon is 100.  One some of my boxes, it is 1000.  On other 2001.

The more I think about it, the more I think that the socat idea you had earlier is most elegant solution possible for this.   We add to the docker-compose a socat service which runs as root and mounts the host docker socket as a volume and proxies it over tcp.  This would make the docker socket permissions problem completely disappear, and as a bonus, it would allow us to easily replace the host docker with dind or with an arbitrary remote docker daemon in a way that would be completely transparent to the Jenkins container.

**Pull request closed without merging**